backend/apis/reports/zipcode-report.api.js

```diff
@@ -0,0 +1,46 @@
+const { app, invalidRequest, pool, databaseError } = require("../../server");
+const { checkValid, nullableValidDate } = require("../utils/validation-utils");
+const mysql = require("mysql");
+const _ = require("lodash");
+const ClientCapability = require("twilio/lib/jwt/ClientCapability");
+
+app.get(`/api/reports/zipcode-report`, (req, res) => {
+  const validationErrors = checkValid(
+    req.query,
+    nullableValidDate("start"),
+    nullableValidDate("end")
+  );
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  const startDate = req.query.start || "2000-01-01T0";
+  const endDate = req.query.end || "3000-01-01T0";
+
+  const sql = mysql.format(
+    `
+        SELECT zip, city, COUNT(*) as clientCount
+        FROM contactInformation GROUP BY zip ORDER BY clientCount DESC;
```

Comment:
The contactInformation table is contains both historical and current rows. Because of this, we have to select only the latest (current) rows from the table.

The following code shows how to do that:

https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/d99947b3edd33a687c56e4bce3719342981574db/backend/apis/clients/get-clients-by-id.sql#L27-L36. In that code, you should copy paste the entire `contactInfo` section (line 27 to 36). The contactInfo temporary table is the one that you can use in this zip code report.

Reply:
Okay. I modified my sql query with the above code. I tested it in the console and it returns correctly.